src/viam/sdk/module/service.cpp

```diff
@@ -133,7 +134,7 @@ ::grpc::Status ModuleService_::ReconfigureResource(
 
     // if the type isn't reconfigurable by default, replace it
     try {
-        res->stop();
+        Stoppable::stop_if_stoppable(res);
```

Comment:
So ... am I reading this right that the only way to reach this code is if `reconfigure` threw an exception that did not derive from `std::exception`?

Also,

Reply:
That sounds correct; my understanding of that code is to call `stop` if the resource does not actually implement a `reconfigure` method (and `throw`s some exception that's not `std::exception`?) and reconstruct the resource instead of reconfiguring.

Reply:
That makes me wonder if we need a `Reconfigurable` mixin, much like `Stoppable`. The status quo is that all `Resource` implementations must implement `reconfigure`, but with an escape hatch where throwing an exception (but only one that doesn't derive from `std::exception`!) is the signal that the resource doesn't actually support reconfiguration. I'm not sure I'm too keen on that design. Maybe a ticket is in order?

Reply:
Yeah I think a `Reconfigurable` class is better. Filed https://viam.atlassian.net/browse/RSDK-5967.